pkg/operator/reporting/validate.go

```diff
@@ -127,12 +105,61 @@ func ValidateGenerationQueryDependencies(deps *ReportGenerationQueryDependencies
 		}
 	}
 
-	if len(errs) != 0 {
-		return fmt.Errorf("ReportGenerationQuery dependency validation error: %s", strings.Join(errs, ", "))
+	if len(validationErr.disabledViewQueryNames) != 0 ||
+		len(validationErr.uninitializedQueryNames) != 0 ||
+		len(validationErr.uninitializedDataSourceNames) != 0 ||
+		len(validationErr.uninitializedReportNames) != 0 ||
+		len(validationErr.uninitializedScheduledReportNames) != 0 {
+		return validationErr
 	}
 	return nil
 }
 
+func IsUninitializedDependencyError(err error) bool {
+	validationErr, ok := err.(*reportGenerationQueryDependenciesValidationError)
```

Comment:
Slightly confused as to how this works. Maybe my golang is a bit rusty?

Reply:
It's a type assertion: https://golang.org/doc/effective_go.html#interface_conversions